lib/parse_date/int_from_string.rb

```diff
@@ -1,7 +1,5 @@
 # frozen_string_literal: true
 
-require 'date' # so upstream callers don't have to require it
```

Comment:
this probably made sense in the context of the stanford-mods gem, but here we are only parsing out years as integers and arrays of integers.  (makes me think I should have named the gem `year_parsing` or `parse_years`)

Reply:
I like removing the unused dependency. I see this gem as probably being useful for expanded date parsing later, so the name is still good, and the dependency can be added back at the time it's needed. So this is a good call to me.